### PR TITLE
Added type annotations

### LINKED ISSUE #31: Add type annotations
Using the typing package/module, begin adding types to function parameters and returns to document the code and to be consistent with the Python community moving towards using types.

### PR DESCRIPTION
**Pull request by gabb222**
Fixes #31

### CODE REVIEW

**Comment by gabb222** [comment]
@Theia-Scientific/developers ready for review

File: tspng/extraction.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def extract_from_file(path, mime_type=MIME_TYPE):
     dict=json.loads(meta[mime_type])
     return dict
 
-def extract_from_files(paths, mime_type=MIME_TYPE):
+def extract_from_files(paths: list, mime_type: str=MIME_TYPE):
```
Comment (volks73) [inline tspng/extraction.py:28]: The `from typing import List` should be added to the top of the file and the `List` "type"/class should be used for the `paths` argument. Using the `list` type is actually a function not a type.

**Review by volks73 — CHANGES_REQUESTED** [review]
Using `list` as a type can lead to weirdness.

For reference, the Python typing module documentation is pretty good: <https://docs.python.org/3.8/library/typing.html>.

File: tspng/extraction.py (hunk 1 above)
Comment (gabb222) [reply tspng/extraction.py]: Changes made as requested

**Comment by gabb222** [comment]
@Theia-Scientific/developers ready for review

File: tspng/extraction.py [hunk 2]
```diff
@@ -1,11 +1,12 @@
 #import statements
 from PIL import Image
 from tspng import MIME_TYPE
+from typing import List
 
 import json
 import os
 
-def extract_from_file(path, mime_type=MIME_TYPE):
+def extract_from_file(path: str, mime_type: str=MIME_TYPE):
```
Comment (volks73) [inline tspng/extraction.py:9]: A type for the return value should be added. This an be done as follows:

```python
def extract_from_file(path: str, mime_type: str = MIME_TYPE) -> Dict:
```

The `Dict` class/type will need to be import from the `typing` module, `from typing import Dict, List`.

File: tspng/extraction.py [hunk 3]
```diff
@@ -25,7 +26,7 @@ def extract_from_file(path, mime_type=MIME_TYPE):
     dict=json.loads(meta[mime_type])
     return dict
 
-def extract_from_files(paths, mime_type=MIME_TYPE):
+def extract_from_files(paths: List, mime_type: str=MIME_TYPE):
```
Comment (volks73) [inline tspng/extraction.py:29]: A type for the return value should be added here, too. I think it can actually be `Dict[Dict]`, but please check on this.

**Review by volks73 — CHANGES_REQUESTED** [review]
Return types can be added to the two functions, too. I missed this on the previous review.

File: tspng/extraction.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
 #import statements
 from PIL import Image
 from tspng import MIME_TYPE
+from typing import List
 
 import json
```
[the remark this answers is not in the document]
Comment (gabb222) [reply tspng/extraction.py:6]: Change made as requested

File: tspng/extraction.py (hunk 2 above)
Comment (gabb222) [reply tspng/extraction.py:9]: Change made as requested

File: tspng/extraction.py (hunk 3 above)
Comment (gabb222) [reply tspng/extraction.py:29]: Change made, Dict[Dict] did not work so for right now it is Dict

**Review by volks73 — APPROVED** [review]
Looks good.

**Pull request merged**